### DISCUSSION TITLE
Question on export templates and jinja

### DISCUSSION THREAD

**Discussion by TimY8976** [netbox-community/netbox#11653] [Q&A]
Hi, I'm getting started with netbox and was planning on calling to netbox for variable info then rendering a config (for juniper box) with jinja. I see that netbox has support for export templates. I'm still working through documentation but can we load a jinja template onto netbox that will spit out say a set config for juniper device? If so, has anyone had luck with it pro/con?

**Comment by zombah** [comment]
Hello,
This was always possible via netbox rest api and ansible for example,
starting with version 3.5 and higher it is also possible without external tools,
check this [blog post](https://netboxlabs.com/blog/how-to-generate-device-configurations-with-netbox) for howto
